lib/metasploit/framework/compiler/mingw.rb

```diff
@@ -18,6 +18,7 @@ def compile_c(src)
           end
 
           stdin_err, status = Open3.capture2e(cmd)
+          cleanup_files
```

Comment:
My only concern with this, and maybe someone could clear this up for me, is that you introduce a timing issue by doing this whereby Open3 tries to run the command in one thread and in this thread we end up racing to clean up the files which may end up occurring before Open3 has finished executing its command.

If this is fine though, I'd recommend also adding it to your other CPP compilation functions so they clean up the CPP files after they are compiled.

Reply:
`Open3` blocks the current thread and runs synchronously:
```ruby
puts Time.now
stdin_err, status = Open3.capture2e("sleep 10")
puts Time.now
```

Which would results in a 10 second delay between the two logs outputs:
```
15:49:16 +0000
15:49:26 +0000
```

Unless I misunderstand the question 👀

Reply:
@adfoster-r7 My bad I was under the impression `Open3` ran asynchronously and did not block the current thread. If that is not the case and it is in fact synchronous then this entire thread is moot and we can mark this conversation as resolved.